### DISCUSSION TITLE
Can we make overwriting .git/config on each push optional?

### DISCUSSION THREAD

**Discussion by working-name** [WeblateOrg/weblate#5441] [Ideas]
Hello!

I'm sure there's a perfectly good explanation for the behavior but in my use case I have not found it to be useful.

It seems weblate overwrites `.git/config` as soon as you hit push making any manual editing or fixing impossible. If I change the file's perm to 400 weblate will crash and send a permission error email.

Possibly unrelated: what are the github username and token from settings.py used for?

**Comment by nijel** [comment]
Weblate writes it exactly to make sure that it matches its configuration. What are you trying to achieve?

> Possibly unrelated: what are the github username and token from settings.py used for?

Did you notice we have a documentation? https://docs.weblate.org/en/latest/admin/config.html#std-setting-GITHUB_USERNAME

**Reply by working-name** [reply]
I was trying to get it to push the changes but it kept coming back with an error about being prompted for a username on a non-interactive terminal. This happened after I removed the initial component and added it again for some other troubleshooting.

I assumed it would utilize the `settings.py` values to push, but it wasn't the case. So I manually edited the url in .git/config, tried using credential helper but didn't realize weblate would overwrite the config file every time.

I finally figured out that I needed to edit the push URL for that component and include credentials there which would in turn be overwritten every time weblate attempts to push changes.

I did RTFM but it didn't click when I read "for pull requests", my bad. Sorry for causing you annoyance as you worked a lot to document everything and here we are, forcing you to answer questions that would be answered by carefully reading the docs.

**Reply by nijel** [reply]
It is recommended to use SSH URLs for accessing repositories, see https://docs.weblate.org/en/latest/vcs.html#accessing-repositories, but you can specify credentials in the HTTPS URL as well, see https://docs.weblate.org/en/latest/vcs.html#https-repositories.
